### PR TITLE
qa/cephfs: remove sudo parameter from run_shell()

### PR DESCRIPTION
**Pull request by rishabh-d-dave**
Right now, run_shell() accepts both "sudo" and "omit_sudo" as parameter,
it's better to accept only for the sake of convention. Later is
apparently better since it's already being accepted by run() methods of
vstart_runner's LocalRemote and teuthology's Remote classes.

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/master/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [x] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Comment by rishabh-d-dave** [comment]
jenkins test api

**Review by nmshelke — APPROVED** [review]
LGTM

File: qa/tasks/cephfs/mount.py [hunk 1]
```diff
@@ -702,17 +702,10 @@ def run_python(self, pyscript, py_version='python3', sudo=False):
 
     def run_shell(self, args, timeout=300, **kwargs):
         kwargs.pop('omit_sudo', False)
-        sudo = kwargs.pop('sudo', False)
         cwd = kwargs.pop('cwd', self.mountpoint)
         stdout = kwargs.pop('stdout', StringIO())
         stderr = kwargs.pop('stderr', StringIO())
 
-        if sudo:
```
Comment (joscollin) [inline qa/tasks/cephfs/mount.py:710]: A better way is to check `omit_sudo` here? Then you could avoid updating run_shell() with `sudo` and then `omit_sudo` they are basically the same thing.

**Comment by rishabh-d-dave** [comment]
@joscollin @nmshelke @neesingh-rh PTAL. Added few more lines of change in mount.py.

**Comment by rishabh-d-dave** [comment]
jenkins test api

**Review by nmshelke — COMMENTED** [review]
Should we also check use of `run_shell` in `../qa/tasks/cephfs/xfstests_dev.py` ?

**Comment by rishabh-d-dave** [comment]
> Should we also check use of `run_shell` in `../qa/tasks/cephfs/xfstests_dev.py` ?

I'll take a look.

**Comment by rishabh-d-dave** [comment]
jenkins test make check

**Comment by rishabh-d-dave** [comment]
jenkins test windows

**Comment by vshankar** [comment]
@rishabh-d-dave Mind running this through some tests via vstart_runner before I pick it up for teuthology tests? (just to catch any _obvious_ bugs/leftovers)

**Comment by rishabh-d-dave** [comment]
> @rishabh-d-dave Mind running this through some tests via vstart_runner before I pick it up for teuthology tests? (just to catch any _obvious_ bugs/leftovers)

@vshankar Sure, I'll run it. I sent you a message on IRC asking the same yesterday or day before.

**Comment by vshankar** [comment]
> > @rishabh-d-dave Mind running this through some tests via vstart_runner before I pick it up for teuthology tests? (just to catch any _obvious_ bugs/leftovers)
> 
> @vshankar Sure, I'll run it. I sent you a message on IRC asking the same yesterday or day before.

ACK. Apologies if I missed replying.

**Comment by vshankar** [comment]
> > > @rishabh-d-dave Mind running this through some tests via vstart_runner before I pick it up for teuthology tests? (just to catch any _obvious_ bugs/leftovers)
> > 
> > 
> > @vshankar Sure, I'll run it. I sent you a message on IRC asking the same yesterday or day before.
> 
> ACK. Apologies if I missed replying.

test_admin via vstart_runner would be good. I'll then pick it up in my test branch.

**Comment by rishabh-d-dave** [comment]
@vshankar 
 
> test_admin via vstart_runner would be good. I'll then pick it up in my test branch.

Done.  It ran successfully  - http://pulpito.front.sepia.ceph.com/rishabh-2022-06-09_04:29:46-fs-main-distro-default-smithi/. I'll try a couple more sub-suites to be extra sure

**Review by vshankar — CHANGES_REQUESTED** [review]
@rishabh-d-dave Needs fixing for these failures - https://pulpito.ceph.com/vshankar-2022-06-19_08:22:46-fs-wip-vshankar-testing1-20220619-102531-testing-default-smithi/6886674/

```
2022-06-19T09:15:27.796 INFO:tasks.cephfs_test_runner:======================================================================
2022-06-19T09:15:27.796 INFO:tasks.cephfs_test_runner:ERROR: test_subvolume_group_quota_legacy_subvolume_removal (tasks.cephfs.test_volumes.TestSubvolumeGroups)
2022-06-19T09:15:27.796 INFO:tasks.cephfs_test_runner:----------------------------------------------------------------------
2022-06-19T09:15:27.797 INFO:tasks.cephfs_test_runner:Traceback (most recent call last):
2022-06-19T09:15:27.797 INFO:tasks.cephfs_test_runner:  File "/home/teuthworker/src/git.ceph.com_ceph-c_4322afc801c50aea640f0a7d6c01cfe9e520b178/qa/tasks/cephfs/test_volumes.py", line 1218, in test_subvolume_group_quota_legacy_subvolume_removal
2022-06-19T09:15:27.797 INFO:tasks.cephfs_test_runner:    self.mount_a.run_shell(['mkdir', '-p', createpath1], sudo=True)
2022-06-19T09:15:27.798 INFO:tasks.cephfs_test_runner:  File "/home/teuthworker/src/git.ceph.com_ceph-c_4322afc801c50aea640f0a7d6c01cfe9e520b178/qa/tasks/cephfs/mount.py", line 717, in run_shell
2022-06-19T09:15:27.798 INFO:tasks.cephfs_test_runner:    omit_sudo=omit_sudo, **kwargs)
2022-06-19T09:15:27.799 INFO:tasks.cephfs_test_runner:  File "/home/teuthworker/src/git.ceph.com_git_teuthology_31a356e0c737f1781079045bb8dc366982be8713/teuthology/orchestra/remote.py", line 510, in run
2022-06-19T09:15:27.799 INFO:tasks.cephfs_test_runner:    r = self._runner(client=self.ssh, name=self.shortname, **kwargs)
2022-06-19T09:15:27.799 INFO:tasks.cephfs_test_runner:TypeError: run() got an unexpected keyword argument 'sudo'
```

**Comment by rishabh-d-dave** [comment]
@vshankar 

Tests are passing now. Following 2 are green -
http://pulpito.front.sepia.ceph.com/rishabh-2022-06-27_15:10:39-fs:functional-wip-vshankar-testing1-20220619-102531-distro-default-smithi
http://pulpito.front.sepia.ceph.com/rishabh-2022-06-27_15:09:58-fs:functional-wip-vshankar-testing1-20220619-102531-distro-default-smithi


volumes: http://pulpito.front.sepia.ceph.com/rishabh-2022-06-27_17:04:18-fs:volumes-wip-vshankar-testing1-20220619-102531-distro-default-smithi

6901211 failed but after all tests ran successfully and 6901214 crashed before running any tests.

**Comment by rishabh-d-dave** [comment]
@vshankar 
> 6901211 failed but after all tests ran successfully and 6901214 crashed before running any tests.

On rerun the first one passed but second continues to fail to run - https://pulpito.ceph.com/rishabh-2022-06-28_04:45:30-fs:volumes-wip-vshankar-testing1-20220619-102531-distro-default-smithi/

**Comment by rishabh-d-dave** [comment]
jenkins test api

**Comment by vshankar** [comment]
@rishabh-d-dave Guessing that this is ready for tests?

**Comment by rishabh-d-dave** [comment]
https://tracker.ceph.com/projects/cephfs/wiki/Main#2022-July-05

**Pull request merged**